src/sql/db_connection_pool/dbconnection/duckdbconn.rs

```diff
@@ -151,10 +151,18 @@ impl SyncDbConnection<r2d2::PooledConnection<DuckdbConnectionManager>, DuckDBPar
                 yield Ok(batch);
             }
 
-            if let Err(e) = join_handle.await {
-                yield Err(DataFusionError::Execution(format!(
-                    "Failed to execute DuckDB query: {e}"
-                )))
+            match join_handle.await {
+                Ok(Err(e)) => {
+                    yield Err(DataFusionError::Execution(format!(
+                        "Failed to execute DuckDB query: {e}"
+                    )))
+                },
+                Err(e) => {
+                    yield Err(DataFusionError::Execution(format!(
+                        "Failed to execute DuckDB query: {e}"
+                    )))
```

Comment:
What is the difference between these cases?

Reply:
`Err(e)` is the `JoinError` from the `JoinHandle`, which would usually trigger if e.g. tokio couldn't spawn the task or something. `Ok(Err(e))` is where tokio has successfully spawned the task, but there's an error from within the task.